### PR TITLE
trlog processing: fetch tenants bug

### LINKED ISSUE #321: Objects "disappear" from S3 bucket but are still available
<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [EdgeFS.io Slack](https://edgefs.slack.com/).
3. Did you already search the existing open issues for anything similar? -->

### Summary 
All objects put into an S3 bucket will disappear from the list of objects in a bucket after a short period of time.  The object will still be available via the S3 service.  The object count in the UI will show 0 objects.

### Steps to reproduce the behavior
- Create new edgefs cluster (3 nodes)
```
# efscli system status -v1
ServerID 42BB489E12429C6E9E20F55CDB48C63A csp-north-test-server3:rook-edgefs-target-1-0 ONLINE
  VDEVID 77DFAB668A96B8A5F9CF89694F54146D scsi-36000c29f0b3a0c2dd3bbb40dc97b9bd4 ONLINE
ServerID F8DE45EDAF76ED0E6DDAB786EE3978A5 csp-north-test-server2:rook-edgefs-target-0-0 ONLINE
  VDEVID 0B93CBEF1D5DF0D2AC739EC62FDB0CC0 scsi-36000c2902c0ec7b5146031394bb4c928 ONLINE
ServerID 77B064CDC2131E8B860C9C92E9D80FB0 csp-north-test-server1:rook-edgefs-target-2-0 ONLINE
  VDEVID 86F64315E27FDD75C551CA74EC0E544B scsi-36000c296e8b5ead27902abdf13bcacc0 ONLINE
```
- Create Cluster/Tenant/Namespace `Hawaii/Cola/bk1`
- Create S3 service
```
# efscli service show s3-cola -s
X-Service-Name: s3-cola
X-Service-Type: s3
X-Service-Worker: s3
X-Servers: -
X-Description: S3 Object
X-Need-MD5: true
X-ACL-On: false
X-List-Cache: true
X-List-All-Buckets: true
X-HTTP-Port: 9982
X-HTTPS-Port: 9443
X-HTTPS-Key: -
X-HTTPS-Cert: -
X-Region: Hawaii
X-List-Max-Size: 1000
X-Ciphers: -
X-Default-Tenant: Cola
X-H2O-Port: -
X-Default-Owner: -
X-Trust-Proxy: true
X-Number-Of-Versions: 1
X-Auth-Type: key_secret
X-Access-Log: 1m,1
X-Status: enabled
X-ContainerIPv6-42BB489E12429C6E9E20F55CDB48C63A: ::ffff:ffff:1d70:100a
X-Container-Hostname-42BB489E12429C6E9E20F55CDB48C63A: csp-north-test-server3
X-ContainerId-42BB489E12429C6E9E20F55CDB48C63A: rook-edgefs-s3-s3-cola-f79c954df-27p47
[
  Hawaii/Cola
]
```
- Create S3 client (using python3/boto3 library) and verify connectivity
```
In [34]: s3.list_buckets()['Buckets']                                                                                                                                                                                                                                                                                                                                       
Out[34]: 
[{'Name': 'bk1',
  'CreationDate': datetime.datetime(2019, 7, 24, 14, 20, 21, 8000, tzinfo=tzutc())}]
```
- Put object into bucket and verify object exists
```
In [35]: s3.put_object(Body=json.dumps(data), Bucket='bk1', Key='test.json')                                                                                                                                                                                                                                                                                                
Out[35]: 
{'ResponseMetadata': {'RequestId': '122d23496e94d648',
  'HostId': '3e5d5f26a35e98bd',
  'HTTPStatusCode': 200,
  'HTTPHeaders': {'x-powered-by': 'Express',
   'x-amz-id-2': '3e5d5f26a35e98bd',
   'x-amz-request-id': '122d23496e94d648',
   'content-length': '0',
   'content-type': 'application/octet-stream',
   'etag': '"42b7b4f2921788ea14dac5566e6f06d0"',
   'date': 'Wed, 24 Jul 2019 14:21:42 GMT',
   'connection': 'keep-alive'},
  'RetryAttempts': 0},
 'ETag': '"42b7b4f2921788ea14dac5566e6f06d0"'}

In [37]: s3.list_objects(Bucket='bk1')['Contents']                                                                                                                                                                                                                                                                                                                          
Out[37]: 
[{'Key': 'test.json',
  'LastModified': datetime.datetime(2019, 7, 24, 14, 8, 16, 430000, tzinfo=tzutc()),
  'ETag': '"42b7b4f2921788ea14dac5566e6f06d0"',
  'Size': 8,
  'StorageClass': 'STANDARD',
  'Owner': {'DisplayName': 'Hawaii_Cola', 'ID': 'Hawaii_Cola'}}]
```
- Wait a short period of time (< 30s) and object will no longer exist
```
In [38]: s3.list_objects(Bucket='bk1')['Contents']                                                                                                                                                                                                                                                                                                                          
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-38-6f70d7a4d9e5> in <module>
----> 1 s3.list_objects(Bucket='bk1')['Contents']

KeyError: 'Contents'
```
- Access missing object
```
In [40]: s3.get_object(Bucket='bk1', Key='test.json')['Body'].read().decode('utf-8')                                                                                                                                                                                                                                                                                        
Out[40]: '{"a": 1}'
```

### Expected behavior
Object listing will reflect all objects in the bucket.

### Actual behavior
Object disappears from list but is still accessible.

### EdgeFS version
1.2.31

### Deployment type

<!--
Select which installation type are you using. 
-->
- [x] Kubernetes with Rook Operator
- [ ] Docker Compose

<!--
Select deployment resource policy. 
-->
- [ ] Embedded with less than 8GB per EdgeFS Target or Daemon container
- [x] Optimal resource allocation with more then 4 CPU and 8GB per EdgeFS Target or Daemon container

### Kubernetes cluster

<!--
What kind of Kubernetes cluster are you using, version and how did you create it?
-->
Kubernetes v1.14.3 deployed via Kubespray
<!--
In which operating system environment are you running Kubernetes cluster?
-->
- [x] RedHat 7.x
- [ ] Ubuntu 16.04.x
- [ ] Ubuntu 18.04.x
- [ ] Other

### PR DESCRIPTION
**Pull request by SergeyKaydalov**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.
- [ ] Change modifies documentation or comments only.

#### Description
Fixed a bug in  trlog_fetch_tenants(). Sometimes it couldn't find tenants started with TRLOG- and thus, couldn't do bucket insert.

#### Which issue this PR fixes

fixes #321

### CODE REVIEW

**Comment by dyusupov** [comment]
:star: PR built and available in a preview environment **nexenta-edgefs-pr-327**

File: src/ccow/src/libtrlog/trlog.c [hunk 1]
```diff
@@ -260,15 +240,18 @@ trlog_fetch_tenants(ccow_t tc, char ***tid_arr, int *tid_arr_len)
 		goto _err;
 	}
 
-	while ((kv = ccow_lookup_iter(iter, CCOW_MDTYPE_NAME_INDEX, pos))) {
-		if (strspn(kv->key, TRLOG_TID_PREFIX) != strlen(TRLOG_TID_PREFIX))
-			break;
+	while ((kv = ccow_lookup_iter(iter, CCOW_MDTYPE_NAME_INDEX, idx))) {
+		printf("TID=%s\n", kv->key);
```
Comment (dyusupov) [inline src/ccow/src/libtrlog/trlog.c]: has to be log_debug() or drop it

Comment (SergeyKaydalov) [reply src/ccow/src/libtrlog/trlog.c]: dropped

**Comment by dyusupov** [comment]
:star: PR built and available in a preview environment **nexenta-edgefs-pr-327**

**Pull request merged**